.gitignore

```diff
@@ -1,8 +1,8 @@
-.cproject
-compile_commands.json
-cscope.*
-tags
-TAGS
-out
-.project
-*.swp
+/.cproject
+/compile_commands.json
+/cscope.*
+/tags
+/TAGS
+/out
+/.project
+/*.swp
```

Comment:
\*.swp\ should apply to all subdirectories.

Reply:
I agree

Reply:
Indeed, but I have considered that ignoring all dot files in the second commit rewrite this ignore rule.
I propose to remove the forward slash in the first commit and then delete the *.swp line in the second one.

Reply:
I would not remove the `*.swp` entry. It was already there and is quite convivient to **vi**-like tools users.

Reply:
Yes of course but the purpose of this PR is to ignore all dot files whatever your preferred text editor. 
To me, keeping the entry `*.swp` alongside the entry `.*` doesn't have so much sense.

Reply:
_*.swp_ files are not dot file. So drop legacy? I would like it to be preserved, but if our dear maintainers thinks it's not worth it, then fine :cry:

Reply:
my bad, there are. Fine :smiley:

Reply:
Note: for first commit to be consistent, it could straight remove the `*.swp` line. `/*.swp` is not very useful to anybody.

Reply:
First step is to add `/` where it makes sense, so NOT on the `*.swp` line. Then the second step adds more `.*` files so it can drop `*.swp` (which should actually have been `.*.swp` because vi does prepend a dot to the file name).